File: app/src/main/java/com/example/androidproject/Users/UserSingleton.java
package com.example.androidproject.Users;


public class UserSingleton {

    private static UserSingleton singleton=null;

    private String currenUserUri;
    private String currentUserName;

    private UserSingleton() {
    }

    public static UserSingleton getInstance() {

        if (singleton == null) {
            synchronized (UserSingleton.class) {
                if (singleton == null) {
                    singleton = new UserSingleton();
                }
            }
        }
        return singleton;
    }

    public void setCurrenUserUri(String currenUserUri) {
        this.currenUserUri = currenUserUri;
    }

    public void setCurrentUserName(String currentUserName) {
        this.currentUserName = currentUserName;
    }

    public String getCurrenUserUri() {
        return currenUserUri;
    }

    public String getCurrentUserName() {
        return currentUserName;
    }
}

File: app/src/main/java/com/example/androidproject/Users/UserSingleton.kt
package com.example.androidproject.Users

import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.ktx.Firebase

class UserSingleton private constructor(){

    lateinit var auth:FirebaseAuth

    init {
        println("Singleton class invoked.")
        auth= FirebaseAuth.getInstance()
    }

    /*
    fun getCurrentUser(): User {
        return currentUser
    }
    
     */

}